src/ZoneServer/Scripting/AI/AiScript.Routines.cs

```diff
@@ -226,7 +262,7 @@ protected IEnumerable Follow(ICombatEntity followTarget, float minDistance = 50,
 
 			while (true)
 			{
-				if (followTarget.Map.Id != this.Entity.Map.Id)
+				if (followTarget == null || followTarget.Map.Id != this.Entity.Map.Id)
```

Comment:
I could be mistaken, but I believe followTarget becomes null if the player logs out.  Prior to adding this change, I noticed that the pets would remain after you logged out and logged back in (essentially becoming cloned), but this change fixes it.

Reply:
Unless something really weird is going on, it should be entirely impossible for this variable to become `null`. It's a parameter of the function, so unless you pass in null, this variable will never be null, because it can't be changed from the outside. And the only location where this function is currently called in Melia is from the basic monster AI, with a null check right before the call. As such, this check seems confusing.

Are you certain this specific check has any kind of actual, meaningful effect?

Reply:
I'll do some more testing on it to see if I can reproduce what I was seeing.